services/src/main/java/io/grpc/protobuf/service/ProtoReflectionService.java

```diff
@@ -0,0 +1,326 @@
+    int extensionNumber = extension.getNumber();
+    if (fileDescriptorsByExtensionAndNumber.containsKey(extensionName)) {
+      fileDescriptorsByExtensionAndNumber.get(extensionName).put(extensionNumber, fd);
+    } else {
+      Map<Integer, FileDescriptor> extensionMap = new HashMap<Integer, FileDescriptor>();
+      extensionMap.put(extensionNumber, fd);
+      fileDescriptorsByExtensionAndNumber.put(extensionName, extensionMap);
+    }
+  }
+
+  private void processService(ServiceDescriptor service, FileDescriptor fd) {
+    fileDescriptorsBySymbol.put(service.getFullName(), fd);
+    for (MethodDescriptor method : service.getMethods()) {
+      fileDescriptorsBySymbol.put(method.getFullName(), fd);
+    }
+  }
+
+  private void processType(Descriptor type, FileDescriptor fd) {
+    fileDescriptorsBySymbol.put(type.getFullName(), fd);
+    for (FieldDescriptor extension : type.getExtensions()) {
+      processExtension(extension, fd);
+    }
+    for (Descriptor nestedType : type.getNestedTypes()) {
+      processType(nestedType, fd);
+    }
+  }
+
+  private void processFileDescriptor(FileDescriptor fd) {
+    fileDescriptorsByName.put(fd.getName(), fd);
+    for (ServiceDescriptor service : fd.getServices()) {
+      processService(service, fd);
+    }
+    for (Descriptor type : fd.getMessageTypes()) {
+      processType(type, fd);
+    }
+    for (FieldDescriptor extension : fd.getExtensions()) {
+      processExtension(extension, fd);
+    }
+  }
+
+  /**
+   * Processes the service file descriptors. This prepares a list of the service names and indexes
+   * the file descriptors by name, symbol, and extension.
+   */
+  private synchronized void initFileDescriptorMaps() {
+    if (mapsInitialized) {
+      // Check if the server's set of services has changed
+      Collection<ServerServiceDefinition> currentServices = server.getServices();
+      HashSet<String> currentServiceNames = new HashSet<String>();
+      for (ServerServiceDefinition service : currentServices) {
+        currentServiceNames.add(service.getServiceDescriptor().getName());
+      }
+      if (serviceNames.equals(currentServiceNames)) {
+        return;
+      }
+    }
+
+    serviceNames = new HashSet<String>();
+    fileDescriptorsByName = new HashMap<String, FileDescriptor>();
+    fileDescriptorsBySymbol = new HashMap<String, FileDescriptor>();
+    fileDescriptorsByExtensionAndNumber = new HashMap<String, Map<Integer, FileDescriptor>>();
+
+    Collection<ServerServiceDefinition> services = server.getServices();
+
+    Queue<FileDescriptor> fileDescriptorsToProcess = new LinkedList<FileDescriptor>();
+    Set<String> seenFiles = new HashSet<String>();
+    for (ServerServiceDefinition service : services) {
+      io.grpc.ServiceDescriptor serviceDescriptor = service.getServiceDescriptor();
+      if (serviceDescriptor.getMarshallerDescriptor() != null
+          && serviceDescriptor.getMarshallerDescriptor()
```

Comment:
break lines at the highest semantic level.  In thi case the instance of should be on the same line as both its args.